dist/ethereum.js

```diff
@@ -2604,39 +2615,65 @@ var requestManager = function() {
         }
 
         // HTTP ASYNC (only when callback is given, and it a HttpProvidor)
-        if(typeof callback === 'function' && provider.name === 'HTTP'){
+        if(typeof callback === 'function' && provider.name === 'RPC over HTTP'){
             provider.send(payload, function(result, status){
 
-                if (!jsonrpc.isValidResponse(result)) {
-                    if(typeof result === 'object' && result.error && result.error.message) {
-                        console.error(result.error.message);
-                        callback(result.error);
-                    } else {
-                        callback(new Error({
-                            status: status,
-                            error: result,
-                            message: 'Bad Request'
-                        }));
-                    }
-                    return null;
+                if(!utils.isArray(data)) {
+                    callback(new Error({
+                        status: status,
+                        error: result,
+                        message: provider.name +' didn\'t repsond with an array of method responses'
+                    }));
+                    return;
                 }
 
-                // format the output
-                callback(null, (typeof data.outputFormatter === 'function') ? data.outputFormatter(result.result) : result.result);
+                result.forEach(function(item, index){
+
+                    if (!jsonrpc.isValidResponse(item)) {
+                        if(typeof item === 'object' && item.error && item.error.message) {
+                            console.error(item.error.message);
+                            callback(item.error);
+                        } else {
+                            callback(new Error({
+                                status: status,
+                                error: item,
+                                message: provider.name +' Bad Request'
+                            }));
+                        }
+                        return null;
```

Comment:
Yes but if its now valid, we want to present either the error we got back, or show at least ageneral erro, if the request failed for other reasons. Though i agree we could put it in an extra funciton. Its only in two places, and won't be anywhere else ever i guess.